### PR TITLE
refactor(planner_manager): cleanup code

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

refactoring new planner manager.

## Tests performed

[[TIER IV INTERNAL LINK]](https://evaluation.tier4.jp/evaluation/reports/00598a6a-b608-59d8-85bf-45b90945b0bf?project_id=prd_jt) This PR 1326/1329
[[TIER IV INTERNAL LINK]](https://evaluation.tier4.jp/evaluation/reports/afbc9e46-7960-5a79-ba55-8de13f4d6843?project_id=prd_jt) Baseline 1326/1329

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/planner_manager.cpp
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/src/planner_manager.cpp:379]: Just for confirmation:
What if there is more than one `waiting_approval_modules`?
For example lane change and avoidance both need approval.

From this code, my understanding is, if avoidance is first found, then `avoidance` will be pushed back to `candidate_module_ptrs`, and once `erase` is called, then both `avoidance` and `lane change` will be removed.

I still haven't checked the actual behavior, but from my understanding, for `enable_simultaneous_execution_as_candidate_module`, this might cause the code to run twice, in order to find lane change. Is my understanding correct?

Comment (satoshi-ota) [reply planning/behavior_path_planner/src/planner_manager.cpp:379]: Thanks.

> From this code, my understanding is, if avoidance is first found, then avoidance will be pushed back to candidate_module_ptrs, and once erase is called, then both avoidance and lane change will be removed.

You're right, but this code is a little bit unintentionally. It should not call clearCandidateModules() twice. (I have to fix it.)

I want to reset candidate modules `candidate_modules_ptrs_` once before some approved modules pop and push them back to `candidate_modules_ptrs_`. 

For example, `candidate_modules_ptrs_` has ext LC right, and `approved_module_ptrs_` has avoidance, normal LC left.

C: `candidate_modules_ptrs_`
A: `approved_module_ptrs_`

Therefore, the expected behavior is,

1. remove and reset all modules in `candidate_modules_ptrs_`. (C: **EMPTY**, A: avoidance, normal LC left)
2. check waiting approval modules in `approved_module_ptrs_`. (both avoidance and normal LC left is waiting approval)
3. pop waiting approval modules in `approved_module_ptrs_`.
4. push them back to `candidate_modules_ptrs_`. (C: avoidance, normal LC left, A: **EMPTY**)

The reason why it should reset `candidate_modules_ptrs_` is that I think it is better to go back to square one and start candidate module's process again in order to remove residual modules that don't consider latest `approved_module_ptrs_` condition. But, there might be ways to work them well without reset.

File: planning/behavior_path_planner/src/planner_manager.cpp
Comment (satoshi-ota) [reply planning/behavior_path_planner/src/planner_manager.cpp:413]: It just a guard for unexpected condition, and I think this condition is never satisfied in nominal case.

File: planning/behavior_path_planner/src/planner_manager.cpp
Comment (satoshi-ota) [reply planning/behavior_path_planner/src/planner_manager.cpp:328]: I'll think better name.

File: planning/behavior_path_planner/src/planner_manager.cpp
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/src/planner_manager.cpp:366]: For consistency of using STL, I would suggest either using 
1. `std::for_each`, or
2. `std::transform` with `std::back_inserter(results)`.

File: planning/behavior_path_planner/src/planner_manager.cpp
Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/src/planner_manager.cpp:379]: > You're right. If avoidance module is waiting approval and is found first in std::find_if, the module and next module (lane change) are removed in erase, even if the next module is also waiting aproval since the lane change module's input (= avoidance module output) maybe change significantly.

Now I understand. Thank you for the explanation. 
Since the intention is a little bit not obvious, I think it is better to put the explanation here as comment.

File: planning/behavior_path_planner/src/planner_manager.cpp
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/src/planner_manager.cpp:372]:
```suggestion
    const auto waiting_approval_modules = [](const auto & m) { return m->isWaitingApproval(); };
```
`this` is not used in this lambda.

File: planning/behavior_path_planner/src/planner_manager.cpp
Comment (satoshi-ota) [reply planning/behavior_path_planner/src/planner_manager.cpp:379]: Thanks. I'll do that :+1:

File: planning/behavior_path_planner/src/planner_manager.cpp
Comment (satoshi-ota) [reply planning/behavior_path_planner/src/planner_manager.cpp]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/3382/commits/66096a0af2bca46194cb7d699d9c90663edff8fe.

File: planning/behavior_path_planner/src/planner_manager.cpp
Comment (satoshi-ota) [reply planning/behavior_path_planner/src/planner_manager.cpp:389]: Added comment in https://github.com/autowarefoundation/autoware.universe/pull/3382/commits/66096a0af2bca46194cb7d699d9c90663edff8fe.

File: planning/behavior_path_planner/src/planner_manager.cpp
Comment (satoshi-ota) [reply planning/behavior_path_planner/src/planner_manager.cpp:334]: I updated the function name `runCandidateModules` to `runRequestModules`.

**Review by kosuke55 — APPROVED** [review]
nice refactor thanks!

**Comment by satoshi-ota** [comment]
I confirmed that there is no critical degration. (Please see the result in description.)

**Pull request merged**
